app/src/main/java/com/randomappsinc/studentpicker/editing/EditNameListFragment.java

```diff
@@ -195,10 +196,18 @@ public void onDeletionSubmitted(String name, int amountToDelete) {
     }
 
     @Override
-    public void onDuplicationSubmitted(String name, int amountToAdd) {
-        nameListDataManager.addName(getContext(), name, amountToAdd, listName);
-        namesAdapter.addNames(name, amountToAdd);
-        UIUtils.showSnackbar(parent, R.string.clones_added);
+    public void onDuplicationSubmitted(String name, int amountToHave, int currentAmount) {
+        if (currentAmount > amountToHave) {
+            int amountToDelete = currentAmount - amountToHave;
+            nameListDataManager.deleteName(getContext(), name, amountToDelete, listName);
```

Comment:
Let's move this logic into NameListDataManager to keep the fragment slim. We can expose a setNameAmount() API.